system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -5827,13 +5827,13 @@ class CLdapClient : implements ILdapClient, public CInterface
         return scopes.length();
     }
 
-    virtual int queryDefaultPermission(ISecUser& user)
+    virtual SecAccessFlags queryDefaultPermission(ISecUser& user)
     {
         const char* basedn = m_ldapconfig->getResourceBasedn(RT_FILE_SCOPE);
         if(basedn == NULL || *basedn == '\0')
         {
             DBGLOG("corresponding basedn is not defined");
-            return -2;
+            return SecAccess_Unavailable;
```

Comment:
does this -2 represent a different secaccess flag than the one used in line 1764, 1789?